spark/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java

```diff
@@ -85,6 +89,58 @@ public boolean concurrentSQL() {
   @Override
   public void close() {}
 
+  String interpolateVariable(String originalCommand) {
+
+    // Pattern allows errors in number of '{' and '}' characters
+    Pattern pattern = Pattern.compile("([^{]*)([{]+[^}]+[}]*).*");
+    Matcher matcher;
+
+    StringBuilder newCommandBuffer = new StringBuilder();
+
+    SparkZeppelinContext sparkZeppelinContext = getSparkInterpreter().getZeppelinContext();
+
+    String residualCommand = originalCommand;
+    // Iterate while there is any pattern left in the residual-command text
+    while ((matcher = pattern.matcher(residualCommand)).matches()) {
+      // Get any characters until the next pattern, add to new-command buffer
+      newCommandBuffer.append(matcher.group(1));
+      // Get the variable pattern
+      String template = matcher.group(2);
+      if (template.matches("[{][^{}][^}]*[}]")) {
+        // Handle variable substitution pattern ...
+        // Extract variable-name from template
+        String variableName = template.substring(1, template.length() - 1);
+        // Get variable-value from interpreter's context
+        Object variableValue = sparkZeppelinContext.get(variableName);
+        if (variableValue == null) {
+          // No such variable - Error!
+          return String.format("ERROR:unknown variable '%s'", variableName);
```

Comment:
as commented, let's not use the return string as the success/error status as well.